### DISCUSSION TITLE
Reload polybar module manually with `polybar-msg`

### DISCUSSION THREAD

**Discussion by fritzrehde** [polybar/polybar#2854] [Q&A]
Is there a command similar to `polybar-msg action reload <BAR> <MODULE>` that would reload a specific module? This would be really useful!

**Comment by patrick96** [comment]
There is not. Modules can't be restarted at all. What exactly do you need it for?

**Comment by fritzrehde** [comment]
Instead of restarting the whole bar using an `sxhkd` shortcut, I would like to reload all modules seperately. The use case would be if I quickly want to see an updated version of one module, which may usually have a long update time (e.g. every 30 seconds). Reloading every module would be visually more appealing than killing and restarting the bar, which looks like a quick flicker.

**Reply by patrick96** [reply]
Hmm, restarting modules is really not feasible right now. We don't even have a way to properly shut down some modules, in the worst case we have to wait for the module to stop sleeping before polybar can shut down. Once I finish #2337, this should get way easier though.

I think force-updating a module may be doable (but not for all modules). I'm guessing you're mainly interested in the script module, right?

**Reply by fritzrehde** [reply]
Yes, nearly all my modules are scripts.
